### ISSUE TITLE
The configuration variable `asm.fcn.signature` is not working properly

### ISSUE THREAD

**Issue by officialcjunior** [rizinorg/rizin#744]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Arch Linux (x64)
| File format of the file you reverse (mandatory)      | ELF
| Architecture/bits of the file (mandatory)            | x86/64
| `rizin -v` full output, **not truncated** (mandatory)  | rizin 0.2.0-git @ linux-x86-64   commit: 2171a7cfbf0e1c75e4164893236efb7ee389db20, build: 2021-03-01__16:58:51

`asm.fcn.signature`, from what the name suggests, should allow the user to choose to display the function signature or the prototype.

### Expected behavior
When set to `true`:
```
┌ (fcn) int main (int argc, char **argv);
│           ; var char *var_8h @ rsp+0x18
│           ; var char *var_10h @ rsp+0x20
```
When set to `false`:
```
┌ (fcn) main;
│           ; var char *var_8h @ rsp+0x18
│           ; var char *var_10h @ rsp+0x20
```
Or something similar.

### Actual behavior

When set to `true`:
```
┌ int main (int argc, char **argv);
│           ; var char *var_8h @ rsp+0x18
│           ; var char *var_10h @ rsp+0x20
```

When set to `false`:
```
┌ (fcn) main 6571
// int main (int argc, char **argv);
│   int main (int argc, char **argv);
│           ; var char *var_8h @ rsp+0x18
│           ; var char *var_10h @ rsp+0x20
```

Here's the output from another binary.
When set to `false`:
```
┌ (fcn) entry0 46
│   entry0 (int64_t arg3);
│ bp: 0 (vars 0, args 0)
│ sp: 0 (vars 0, args 0)
│ rg: 1 (vars 0, args 1)
│           0x00005b20      endbr64
```

When set to `true`:
```
┌ entry0 (int64_t arg3);
│ bp: 0 (vars 0, args 0)
│ sp: 0 (vars 0, args 0)
│ rg: 1 (vars 0, args 1)
│           0x00005b20      endbr64
```
### Steps to reproduce the behavior
- Open up a binary, seek to a function and toggle the variable  `asm.fcn.signature`

**Comment by Yuuoniy** [comment]
I think the mistake code lies in `ds_show_functions in disasm.c ` ,  variable `asm.calls` also has a similar problem.  
`asm.fcn.signature` and `asm.calls` are both  related to function signature print. their values are not handled properly.

**Comment by Yuuoniy** [comment]
I am working on this issue. There are something I want to make clear. 
1. Does rizin needs to print function size after function name/signature? (just as now) . Or just as the Expected behavior the issue mentions, get rid of function size.
2. What if `asm.fcn.signature=false `while `asm.calls=true ` what's the expected output ?  As `asm.calls` is used for `showing callee function related info as comments in disasm` , so it may output function signature as comments.

**Comment by ITAYC0HEN** [comment]
Hey! Thanks for taking this one :)
There is no need to print the size in any case, unless `asm.fcn.size` is enabled. And if it does, then it should be prior to the printed function signature and not in its end
See: https://github.com/rizinorg/rizin/pull/669

**Comment by Yuuoniy** [comment]
Maybe you can close this issue, this PR #820 resolve is.

**Comment by XVilka** [comment]
@Yuuoniy thanks! Next time you could just put `Closes #35443435` in the pull request description (see the template) and it will be done automatically.
